ospf6d/ospf6_asbr.c

```diff
@@ -2316,6 +2316,37 @@ void ospf6_asbr_redistribute_reset(struct ospf6 *ospf6)
 	}
 }
 
+void ospf6_asbr_redistribute_reconfigure(struct ospf6 *ospf6)
+{
+	int type;
+	struct ospf6_redist *red;
+	char *rmap_name = NULL;
+
+	for (type = 0; type <= ZEBRA_ROUTE_MAX; type++) {
+		if (type == ZEBRA_ROUTE_OSPF6)
+			continue;
+		red = ospf6_redist_lookup(ospf6, type, 0);
+		if (!red)
+			continue;
+
+		if (type == DEFAULT_ROUTE) {
+			ospf6_redistribute_default_set(
+				ospf6, ospf6->default_originate);
+			continue;
+		}
+		if (ROUTEMAP_NAME(red)) {
+			rmap_name = malloc(sizeof(red));
```

Comment:
This looks like a case of memory leak to me.

Reply:
Why are you allocating and then overriding with another allocated piece of memory from strdup()?

Reply:
Also did we need to malloc it? Could we not stored it temporarily in another local variable?

Reply:
I've removed the malloc because strdup does a malloc

Reply:
But we do not need to use strdup here. A temporary buffer on local stack should suffice. The malloc as part of strdup() will add unnecessary memory management overhead. Just use a local variable and strlcpy() instead of strdup().